streams/src/main/java/org/apache/kafka/streams/state/internals/Segments.java

```diff
@@ -45,15 +45,24 @@
     private long maxSegmentId = -1L;
 
     Segments(final String name, final long retentionPeriod, final int numSegments) {
+        if (numSegments < 1) {
+            throw new IllegalArgumentException("numSegments must be positive");
+        }
+        if (retentionPeriod < 1) {
+            throw new IllegalArgumentException("retentionPeriod must be positive");
+        }
         this.name = name;
         this.numSegments = numSegments;
-        this.segmentInterval = Math.max(retentionPeriod / (numSegments - 1), MIN_SEGMENT_INTERVAL);
+        this.segmentInterval = numSegments == 1 ? Long.MAX_VALUE : Math.max(retentionPeriod / (numSegments - 1), MIN_SEGMENT_INTERVAL);
```

Comment:
@mjsax you can use `Stores.create(..)...` or you can use the `RocksDBStoreSupplier` . But yeah we could make it a min of 2 unless we think there is valid reason why you'd want a window store with a single segment.